joomla/joomla-cms
# HTML5 pattern in validate.js

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/12857 .

### Summary of Changes
HTML5 NATIVE validation is done through `pattern="something"`
The validate.js is adjusted so if pattern tag exists will take precedence else we fall back to the rules

### Testing Instructions
- try adding any of the patterns to a text, email, etc field and see if this works
- use http://html5pattern.com/Names for reference

Add `pattern="[a-zA-Z]+"` after https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_content/models/forms/article.xml#L13
Create a new article and try to submit without a title or with title as 87587676 (or anything apart english alphabet letters)

![screen shot 2016-12-05 at 16 35 31](https://cloud.githubusercontent.com/assets/3889375/20888847/2fd82b8e-bb09-11e6-9499-66579720c686.png)

@Fedik @ggppdk @mbabker


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/joomla/joomla-cms/13094)
